runtime/src/main/java/org/corfudb/runtime/view/ObjectBuilder.java

```diff
@@ -92,6 +95,11 @@ public T open() {
             streamID = CorfuRuntime.getStreamID(streamName);
         }
 
+        // Some Corfu Collections like CorfuQueue need globally ordered UUIDs
+        if (needOrderedUUIDs) {
```

Comment:
This can be instrumented from a higher layer. The object view doesn't need to know whether the uuid are ordered or not, it just needs a uuid.

Reply:
Done.